lib/puppet/functions/foreman_proxy_content/path_from_url.rb

```diff
@@ -0,0 +1,32 @@
+require 'uri'
+
+# @summary Parses a URL and returns the path
+#
+# @example Calling the function
+#   $path = foreman_proxy_content::path_from_url('https://test.example.com/meme_folder')
+#   Returns '/meme_folder'
+#
+# @example Calling the function
+#   $path = foreman_proxy_content::path_from_url('https://test.example.com/meme_folder/funny_meme.jpg')
+#   Return '/meme_folder/funny_meme.jpg'
+#
+# @example Calling the function
+#   $path = foreman_proxy_content::path_from_url('https://test.example.com/')
+#   Returns '/'
+#
+# @example Calling the function
+#   $path = foreman_proxy_content::path_from_url('http://test.example.com')
+#   Returns ''
+#
+Puppet::Functions.create_function(:'foreman_proxy_content::path_from_url') do
+  # @param url URL to parse
+  # @return The path found within the URL
+  dispatch :path do
+    param 'Stdlib::HTTPUrl', :url
+    return_type 'String'
```

Comment:
Didn't find a suitable type for URL path but I could update this with a `Pattern`.

The regex that `URI` uses frustratingly has comments resembling named capture groups, but not the actual names:

```
irb(main):001:0> URI.regexp
=> 
/
        ([a-zA-Z][\-+.a-zA-Z\d]*):                           (?# 1: scheme)
        (?:
           ((?:[\-_.!~*'()a-zA-Z\d;?:@&=+$,]|%[a-fA-F\d]{2})(?:[\-_.!~*'()a-zA-Z\d;\/?:@&=+$,\[\]]|%[a-fA-F\d]{2})*)                    (?# 2: opaque)
        |
           (?:(?:
             \/\/(?:
                 (?:(?:((?:[\-_.!~*'()a-zA-Z\d;:&=+$,]|%[a-fA-F\d]{2})*)@)?        (?# 3: userinfo)
                   (?:((?:(?:[a-zA-Z0-9\-.]|%\h\h)+|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}|\[(?:(?:[a-fA-F\d]{1,4}:)*(?:[a-fA-F\d]{1,4}|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})|(?:(?:[a-fA-F\d]{1,4}:)*[a-fA-F\d]{1,4})?::(?:(?:[a-fA-F\d]{1,4}:)*(?:[a-fA-F\d]{1,4}|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}))?)\]))(?::(\d*))?))? (?# 4: host, 5: port)
               |
                 ((?:[\-_.!~*'()a-zA-Z\d$,;:@&=+]|%[a-fA-F\d]{2})+)                 (?# 6: registry)
               )
             |
             (?!\/\/))                           (?# XXX: '\/\/' is the mark for hostport)
             (\/(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*(?:;(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*)*(?:\/(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*(?:;(?:[\-_.!~*'()a-zA-Z\d:@&=+$,]|%[a-fA-F\d]{2})*)*)*)?                    (?# 7: path)
           )(?:\?((?:[\-_.!~*'()a-zA-Z\d;\/?:@&=+$,\[\]]|%[a-fA-F\d]{2})*))?                 (?# 8: query)
        )
        (?:\#((?:[\-_.!~*'()a-zA-Z\d;\/?:@&=+$,\[\]]|%[a-fA-F\d]{2})*))?                  (?# 9: fragment)
      /x
irb(main):002:0> URI.regexp.names
=> []
```